test/gtest/ucs/test_async.cc

```diff
@@ -349,10 +375,46 @@ class test_async_mt : public test_async {
         return result;
     }
 
-    void spawn() {
+    int repeat_create_destroy_run(unsigned index) {
+        static_assert(std::is_base_of<local_event, LOCAL>::value, "");
+        LOCAL *le;
+        m_ev[index] = le = new LOCAL(GetParam());
```

Comment:
Is the only reason for using `new` here is that `LOCAL` is template? If so, I think can be eliminated by templating the function.

Reply:
Then why use `new` at all, and not alloc on stack?